vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/action/ViewEditHistoryEpoxyController.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package im.vector.riotx.features.home.room.detail.timeline.action
+
+import android.content.Context
+import android.text.Spannable
+import android.text.format.DateUtils
+import androidx.core.content.ContextCompat
+import com.airbnb.epoxy.TypedEpoxyController
+import com.airbnb.mvrx.Fail
+import com.airbnb.mvrx.Incomplete
+import im.vector.matrix.android.api.session.events.model.Event
+import im.vector.matrix.android.api.session.events.model.toModel
+import im.vector.matrix.android.api.session.room.model.message.MessageTextContent
+import im.vector.riotx.R
+import im.vector.riotx.core.extensions.localDateTime
+import im.vector.riotx.core.ui.list.genericFooterItem
+import im.vector.riotx.core.ui.list.genericItem
+import im.vector.riotx.core.ui.list.genericItemHeader
+import im.vector.riotx.core.ui.list.genericLoaderItem
+import im.vector.riotx.features.home.room.detail.timeline.helper.TimelineDateFormatter
+import im.vector.riotx.features.html.EventHtmlRenderer
+import me.gujun.android.span.span
+import name.fraser.neil.plaintext.diff_match_patch
+import timber.log.Timber
+import java.util.*
+
+/**
+ * Epoxy controller for reaction event list
+ */
+class ViewEditHistoryEpoxyController(private val context: Context,
+                                     val timelineDateFormatter: TimelineDateFormatter,
+                                     val eventHtmlRenderer: EventHtmlRenderer) : TypedEpoxyController<ViewEditHistoryViewState>() {
+
+    override fun buildModels(state: ViewEditHistoryViewState) {
+        when (state.editList) {
+            is Incomplete -> {
+                genericLoaderItem {
+                    id("Spinner")
+                }
+                return
+            }
+            is Fail       -> {
+                genericFooterItem {
+                    id("failure")
+                    text(context.getString(R.string.unknown_error))
+                }
+                return
+            }
+        }
+        val sourceEvents = state.editList() ?: return
```

Comment:
use `is Success` in the previous `when` to avoid doing this useless check, and remove the 2 `return` statements for the other cases